### ISSUE TITLE
Long texts take a long time to open

### ISSUE THREAD

**Issue by gerroon** [zadam/trilium#798]
Hi

The longer the note gets longer it takes to show it. 

For instance web clip this page for testing. You will see it takes some seconds to open it. This gets words if the user has much longer texts, which I do. I sometimes wait 5-10 secs for a page to show up.

I am mainly on Linux.

https://docs.blender.org/api/current/bmesh.ops.html

**Comment by zadam** [comment]
Yes, it is indeed very slow to load. This is caused by CKEditor and I can't really do much about it. However, it looks like CKEditor team will look into performance issues soon enough: https://github.com/ckeditor/ckeditor5/issues/5880

I attached this page as an example to that ticket.

**Comment by gerroon** [comment]
Would having read only and edit modes help in this case? For instance Trilium could turn read only mode for large texts by default

**Comment by gerroon** [comment]
It seems like this works only in the desktop version. Just tested it with the server version on Firefox 74 and long text are still very painful there, not sure if this is a package limitation.

**Comment by gerroon** [comment]
@zadam  

Is it possible to add this fix to all other text types like markdown and regular text? Although they are not as bad as the regular text type but can still use the same trick.

**Comment by zadam** [comment]
Specifically for code notes there would be a loss of functionality since no editor means no syntax highlighting ...

**Comment by gerroon** [comment]
I am guessing that you mean the view only mode?

**Comment by zadam** [comment]
Yes, the view only mode would just display plain text (that's how code notes are stored) with no syntax highlighting.

It is possible to include (faster) read only highlighter but there would be probably differences in exact highlighting style.

All this is mostly non-problem for "text" notes since CKEditor produces HTML and also provides "content styles" so read only view can then look very similar to how the content looks in editor mode.

**Comment by gerroon** [comment]
Personally I prefer performance over the looks, and given that only huge files will be effected by this it might be ok.

**Comment by zadam** [comment]
Yeah, makes sense. I added the preview mode to code notes as well (will be in 0.42).
